### PR TITLE
watchman 3.7.0

### PR DESCRIPTION
**Pull request by wez**
We tagged this today.

Tested via:

```
brew install watchman
brew audit watchman
brew test watchman
```

### CODE REVIEW

File: Library/Formula/watchman.rb [hunk 1]
```diff
@@ -2,9 +2,9 @@ class Watchman < Formula
   desc "Watch files and take action when they change"
   homepage "https://github.com/facebook/watchman"
   head "https://github.com/facebook/watchman.git"
-  url "https://github.com/facebook/watchman/archive/v3.3.0.tar.gz"
-  sha256 "c8bd6b496f5d86b13c91368bf3b01346282f565c8af4f8e2f7bb6b74a48c4793"
-
+  url "https://github.com/facebook/watchman/archive/v3.7.0.tar.gz"
+  sha256 "dc0afbe2f64a80926325a61428e3d69d4412f9093a8119db26fa26f7b599bd07"
+
```
Comment (DomT4) [inline Library/Formula/watchman.rb]: Can you kill the whitespacing on line 7 please? Our test-bot will flag it.

**Comment by wez** [comment]
done!

**Comment by dunn** [comment]
It's not building on Mountain Lion:
```
ioprio.c:32:30: error: use of undeclared identifier 'IOPOL_STANDARD'
      low ? IOPOL_THROTTLE : IOPOL_STANDARD);
                             ^
1 error generated.
make[1]: *** [watchman-ioprio.o] Error 1
```
http://bot.brew.sh/job/Homebrew%20Pull%20Requests/30884/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/install_watchman/

**Comment by DomT4** [comment]
Can we place bets on this being a `cxx11` issue? :moneybag: :moneybag:

**Comment by wez** [comment]
https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man3/getiopolicy_np.3.html says that 10.5 introduced the io priority stuff :-/

I can't see whatever that links points to without increasing the level of access that you have to my github account; is there a public visible version of that information?

**Comment by DomT4** [comment]
> I can't see whatever that links points to without increasing the level of access that you have to my github account

We don't have control over the requested permission level FWIW, that's sadly out of our hands.

Alex printed about as much information as there is available:

```
Your build configuration:

        CPPFLAGS =  -D_REENTRANT -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
        CFLAGS = -g -O2 -I/usr/local/Cellar/pcre/8.37/include -Wall -Wextra -Wdeclaration-after-statement -g -gdwarf-2 -fno-omit-frame-pointer
        LDFLAGS = 
        prefix: /usr/local/Cellar/watchman/3.7.0
        version: 3.7.0

==> make
/usr/bin/make  all-am
  CC       watchman-argv.o
  CC       watchman-envp.o
  CC       watchman-spawn.o
  CC       watchman-opt.o
  CC       watchman-cfg.o
  CC       watchman-fstype.o
  CC       watchman-log.o
  CC       watchman-json.o
  CC       watchman-bser.o
  CC       watchman-hash.o
  CC       watchman-ht.o
  CC       watchman-ioprio.o
  CC       watchman-listener.o
  CC       watchman-clientmode.o
  CC       watchman-main.o
ioprio.c:32:30: error: use of undeclared identifier 'IOPOL_STANDARD'
      low ? IOPOL_THROTTLE : IOPOL_STANDARD);
                             ^
1 error generated.
make[1]: *** [watchman-ioprio.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
```

**Comment by wez** [comment]
ok, based on this in `sys/resource.h`:

```
/* I/O Priority */
#define IOPOL_DEFAULT   0
#define IOPOL_IMPORTANT   1
#define IOPOL_PASSIVE   2
#define IOPOL_THROTTLE    3
#define IOPOL_UTILITY   4
#define IOPOL_STANDARD    5

/* compatibility with older names */
#define IOPOL_APPLICATION       IOPOL_STANDARD
#define IOPOL_NORMAL            IOPOL_IMPORTANT
```

I'm guessing that we need to use `IOPOL_APPLICATION` on Mountain Lion.

I'll see if I can finagle that into something we can build here

**Comment by mikemcquaid** [comment]
> I can't see whatever that bot.brew.sh link points to without increasing the level of access that you have to my github account; is there a public visible version of that information?

We only ask to read public information and the organisations you are a member of for authentication. There is not a public version available. Reading test output is required to contribute to Homebrew, sorry.

**Comment by wez** [comment]
If those tests pass, I'll cut a proper tag and re-update; please don't merge this as-is

**Comment by DomT4** [comment]
@wez Seemed to work okay.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/fccaf89e6df9058400f27305ab2023ed6887cbea. Thank you for your contribution to Homebrew @wez!

**Pull request closed without merging**